### ISSUE TITLE
Suggestion: Train on uploaded data

### ISSUE THREAD

**Issue by ruess** [code-debug228/HostetskiGPT#11]
Hey there,

It would be really cool and make this tool way more useful I think if we could create a private/closed model in GPT that is trained on our data. I think I've mentioned this in another issue, but here would be the guide for how to do this:
https://openai.com/blog/customizing-gpt-3

As an example, here's a new product that does effectively that: scanning your entire website to build a private model that you can help to train:
https://sms-bot.vercel.app/

Thanks!

**Comment by Sim0nW0lf** [comment]
Apart from a website, it would be helpful to feed it with saved replies from the freescout plugin too so that previous answers will eventually end up in ChatGPT responses as well.

**Comment by ruess** [comment]
Yes - great idea @Sim0nW0lf  - this is how it will really be able to serve relevant and accurate responses.

**Comment by code-debug228** [comment]
Hello, I'm slowly experimenting with GPT learning capabilities, so in the future this functionality will most likely appear in the module. Thanks for your idea.

**Comment by ruess** [comment]
Thanks for the update @code-debug228 - very exciting!

**Comment by zackkatz** [comment]
Being trained on existing data would be ideal. A new service came out that does this with Help Scout data: https://www.supportagent.ai

**Comment by AliakseyM** [comment]
> Being trained on existing data would be ideal. A new service came out that does this with Help Scout data: https://www.supportagent.ai

If we were to charge users $20 per support agent and an additional $20 for every 100 responses generated by ChatGPT, similar to the service you mentioned, it is possible that implementing such a feature would be faster. However, at the moment, we are supporting the product for free, and it will take some time to add this feature.

**Comment by zackkatz** [comment]
lol, yeah. No expectations 😆 By the way, I'm planning on contributing—not just outputting unreasonable expectations :-)
